app/controllers/measurements_controller.rb

```diff
@@ -2,47 +2,50 @@ class MeasurementsController < ApplicationController
   layout false
 
   def show
-    iteration_time = reference_time - params[:iteration].to_i * interval
-    cache_key = {
-      id: params[:id],
-      day: iteration_time.yday,
-      interval_number: interval_number(iteration_time)
-    }
-
-    @readings = Rails.cache.fetch(cache_key, expires_in: 25.hours) do
-      @measurement = Measurement.find(params[:id])
-      @measurement.
-        readings.
-        includes(:sensor).
-        where('time > ? AND time <= ?', from, to).
-        to_json(include: :sensor)
+    if params[:iteration].blank?
+      last_reading = Reading.where(measurement_id: params[:id]).last
```

Comment:
I was thinking about adding order here, but then i will be a little bit slower (of course I can add it:)). Right now last created reading will be returned.